vertx-auth-oauth2/src/main/java/io/vertx/ext/auth/oauth2/dcr/impl/KeycloakClientRegistrationImpl.java

```diff
@@ -0,0 +1,62 @@
+package io.vertx.ext.auth.oauth2.dcr.impl;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.impl.http.SimpleHttpClient;
+import io.vertx.ext.auth.impl.http.SimpleHttpResponse;
+import io.vertx.ext.auth.oauth2.DCROptions;
+import io.vertx.ext.auth.oauth2.DCRRequest;
+import io.vertx.ext.auth.oauth2.DCRResponse;
+import io.vertx.ext.auth.oauth2.dcr.KeycloakClientRegistration;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public final class KeycloakClientRegistrationImpl implements KeycloakClientRegistration {
+
+  private final Vertx vertx;
+  private final SimpleHttpClient simpleHttpClient;
+
+  private final DCROptions dcrOptions;
+
+  public KeycloakClientRegistrationImpl(Vertx vertx, DCROptions dcrOptions) {
+    this.vertx = vertx;
+    this.dcrOptions = dcrOptions;
+    this.simpleHttpClient = new SimpleHttpClient(vertx, "dcr-client", dcrOptions.getHttpClientOptions());
+  }
+
+  @Override
+  public DCRResponse create(String clientId) {
+    final JsonObject payload = new JsonObject(Map.of("clientId", clientId));
+    final Future<SimpleHttpResponse> response = simpleHttpClient.fetch(HttpMethod.POST,
+      dcrOptions.getResourceUri(), dcrOptions.getInitialAccessToken(), payload.toBuffer());
+    final SimpleHttpResponse simpleHttpResponse;
+    try {
+      simpleHttpResponse = response.await(10, TimeUnit.SECONDS);
+    } catch (TimeoutException e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
We don't use this pattern, the expectation is to use only futures.

You should do something like:

```java
return response.compose(simpleHttpResponse -> {
  // do some validation logic ...
  return new Future.succeededFuture(new DCRResponse(...))
});
```

Reply:
If there is a need to deal with exceptions, you can always chain a

```java
.onFailure(ex -> { ... });
```

In this case we always let it bubble to the receiver, so no need to handle errors. Unless we want to have custom types, try to recover, etc...